src/broker/broker.c

```diff
@@ -37,6 +41,10 @@ static int broker_dispatch_signals(DispatchFile *file) {
 
         c_assert(l == sizeof(si));
 
+        if (si.ssi_signo == SIGHUP) {
+                broker_propagate_sighup(broker);
```

Comment:
You probably want to propagate the return value of this call via error_fold(). You currently fall-through to DISPATCH_EXIT, which will terminate the broker (which likely is what you are seeing in your tests).